src/journal/logic.py

```diff
@@ -215,6 +218,12 @@ def handle_unassign_issue(request, article, issues):
         if issue_to_unassign in issues:
             issue_to_unassign.articles.remove(article)
             issue_to_unassign.save()
+            if issue_to_unassign == article.primary_issue:
+                if article.issues_list().exists():
+                    article.primary_issue = article.issues_list().first()
+                else:
+                    article.primary_issue = None
+                article.save()
```

Comment:
This is good, however, it might be better to handle this in the Signal we already have listening to articles being added and removed from `issue.articles`, that way even if its done in the admin area this will be executed. The signal is: `journal.models.issue_articles_change`